pyo3-macros-backend/src/pyclass.rs

```diff
@@ -445,7 +498,58 @@ fn impl_enum_class(
         }
     };
 
-    let default_impls = gen_default_slot_impls(cls, vec![default_repr_impl]);
+    let repr_type = &enum_.repr;
+
+    let default_int = {
+        // This implementation allows us to convert &T to #repr_type without implementing `Copy`
+        let variants_to_int = variants.iter().map(|variant| {
+            let variant_name = variant.ident;
+            quote! { #cls::#variant_name => #cls::#variant_name as #repr_type, }
+        });
+        quote! {
+            #[doc(hidden)]
+            #[allow(non_snake_case)]
+            #[pyo3(name = "__int__")]
+            fn __pyo3__int__(&self) -> #repr_type {
+                match self {
+                    #(#variants_to_int)*
+                }
+            }
+        }
+    };
+
+    let default_richcmp = {
+        let variants_eq = variants.iter().map(|variant| {
+            let variant_name = variant.ident;
+            quote! {(#cls::#variant_name, #cls::#variant_name) => Ok(true.to_object(py)),}
+        });
+        quote! {
+            #[doc(hidden)]
+            #[allow(non_snake_case)]
+            #[pyo3(name = "__richcmp__")]
+            fn __pyo3__richcmp__(&self, py: ::pyo3::Python, other: &PyAny, op: ::pyo3::basic::CompareOp) -> PyResult<PyObject> {
+                match op {
+                    ::pyo3::basic::CompareOp::Eq => {
```

Comment:
I'm afraid you'll need to rebase on main now that #2022 has merged

Reply:
In particular, `::pyo3` needs to become `_pyo3`.